### ISSUE TITLE
v4 a2 Regression on alwaysOn filters with empty string

### ISSUE THREAD

**Issue by smeng9** [marmelab/react-admin#7191]
**What you were expecting:**

The list is supposed to render there, it should have the same result as no filter is applied
![Screenshot from 2022-02-04 17-53-02](https://user-images.githubusercontent.com/38666763/152618379-90fc6db9-2608-41f9-a40f-e8d1418d5f0c.png)

**What happened instead:**

All the list get filtered out, even if the filter is not set
![Screenshot from 2022-02-04 17-52-26](https://user-images.githubusercontent.com/38666763/152618393-576a9ef4-7b9e-4e4b-ad45-742b146f7edd.png)

**Steps to reproduce:**
1. Fix the dependency react-hook-form of example/totorial in next branch
2. Add `<TextInput source="title" label="Title" alwaysOn />` to postFilters
3. Make a clean render of the page (make sure no encoded filters in url)
4. Go to posts collection through the menu
5. The page will say no results found

This is supposed to be a bug because if a filter input is empty as the initial state, it should not be included in the filter.
Otherwise everything will be filtered out.

However, after type in something and the remove everything typed in the TextInput filter. The list view seems to work correctly.
This is a user experience regression.

**Related code:**

```
https://codesandbox.io/s/amazing-http-0ibjn?file=/src/posts.js:300-355
```

**Other information:**

**Environment**

* React-admin version: 4.0.0-alpha.2
* Last version that did not exhibit the issue (if applicable): 3.19.7
* React version: 17.0.2
* Browser: FireFox 96
* Stack trace (in case of a JS error):

**Comment by WiXSL** [comment]
Reproduced. Thanks

**Comment by fzaninotto** [comment]
Fixed by #7301
